alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/spring/integration/akka/SagaDataExtension.java

```diff
@@ -21,80 +21,84 @@
-
-    //private final ConcurrentLinkedQueue<String> globalTxIds = new ConcurrentLinkedQueue<>();
+    private SagaData lastSagaData;
     private final ConcurrentHashMap<String, SagaData> sagaDataMap = new ConcurrentHashMap();
-    private String lastGlobalTxId;
-    private CleanMemForTest cleanMemForTest = new CleanMemForTest(sagaDataMap);
-    private volatile ApplicationContext applicationContext;
     private MetricsService metricsService;
-
-    public SagaDataExt() {
-      // Just to avoid the overflow of the OldGen for stress testing
-      // Delete after SagaData persistence
-      if (autoCleanSagaDataMap) {
-        new Thread(cleanMemForTest).start();
-      }
-    }
+    private TransactionRepositoryChannel repositoryChannel;
 
     public void putSagaData(String globalTxId, SagaData sagaData) {
-      //if(!globalTxIds.contains(globalTxId)){
-      lastGlobalTxId = globalTxId;
-      //  globalTxIds.add(globalTxId);
-      //}
       sagaDataMap.put(globalTxId, sagaData);
+      lastSagaData = sagaData;
     }
 
     public void stopSagaData(String globalTxId, SagaData sagaData) {
-      // TODO save SagaDate to database and clean sagaDataMap
       this.putSagaData(globalTxId, sagaData);
-      lastGlobalTxId = globalTxId;
       if (sagaData.getLastState() == SagaActorState.COMMITTED) {
         this.metricsService.metrics().doCommitted();
       } else if (sagaData.getLastState() == SagaActorState.COMPENSATED) {
         this.metricsService.metrics().doCompensated();
       } else if (sagaData.getLastState() == SagaActorState.SUSPENDED) {
         this.metricsService.metrics().doSuspended();
       }
+      List<SagaSubTransaction> subTransactions = new ArrayList();
+      sagaData.getTxEntityMap().forEach((k,v)->{
+        subTransactions.add(SagaSubTransaction.builder()
+            .parentTxId(v.getParentTxId())
+            .localTxId(v.getLocalTxId())
+            .beginTime(v.getBeginTime())
+            .endTime(v.getEndTime())
+            .state(v.getState())
+            .build());
+      });
+      GloablTransaction record = GloablTransaction.builder()
+          .type(TransactionType.SAGA)
+          .serviceName(sagaData.getServiceName())
+          .instanceId(sagaData.getInstanceId())
+          .globalTxId(sagaData.getGlobalTxId())
+          .beginTime(sagaData.getBeginTime())
+          .endTime(sagaData.getEndTime())
+          .state(sagaData.getLastState())
+          .subTxSize(sagaData.getTxEntityMap().size())
+          .subTransactions(subTransactions)
+          .events(sagaData.getEvents())
+          .build();
+      repositoryChannel.send(record);
+      sagaDataMap.remove(globalTxId);
     }
 
-    public SagaData getSagaData(String globalTxId) {
-      // TODO If globalTxId does not exist in sagaDataMap then
-      //  load from the database
-      return sagaDataMap.get(globalTxId);
+    // Only for Test
+    public SagaData getLastSagaDate() {
```

Comment:
If this method is only for test, then it should be in test code.
But I'm not familiar with Actor system, so I can not give any advice, sorry.

Besides, typo, it should be `getLastSagaData()`